### PR TITLE
OR: Token Updates as a prerequisite for including integration tests.

### PR DESCRIPTION
**Pull request by a-stacey**
Update token code so that we can pass a keyset to verify signatures against and generate a keyset for use in tests. This is a partial PR for #385, have a commit for commenting the tests back in locally, but separating the PRs to keep the size down and as re including the tests also identified some bugs that is also part of the PR that comments the tests back in.

This is now a huge PR, can break it down into smaller PR's if needed, but the individual commits are fairly self contained and so you can review mostly by using individual commits if needed.

### CODE REVIEW

File: projects/mirza-common-haskell/src/Mirza/Common/GS1BeamOrphans.hs [hunk 1]
```diff
@@ -263,8 +263,8 @@ instance FromField EPC.GS1CompanyPrefix where
 instance ToField EPC.GS1CompanyPrefix where
   toField (EPC.GS1CompanyPrefix pfx) = toField pfx
 
-gs1CompanyPrefixType :: BMigrate.DataType PgDataTypeSyntax EPC.GS1CompanyPrefix
-gs1CompanyPrefixType = textType
+gs1CompanyPrefixFieldType :: BMigrate.DataType PgDataTypeSyntax EPC.GS1CompanyPrefix
```
Comment (sajidanower23) [inline projects/mirza-common-haskell/src/Mirza/Common/GS1BeamOrphans.hs]: Wondering why the change in name here?

File: projects/or_scs/src/Mirza/OrgRegistry/Handlers/Keys.hs [hunk 2]
```diff
@@ -206,52 +211,42 @@ keyStateQuery kid = do
...
   -- logically and progressively builds constraints in order and so that the correct error message is shown when the
   -- keyId is not found, rather then failing because the user id check in doesUserOwnKeyQuery fails.
-  key <- getKeyById keyId
-  when (isNothing key) $ throwing _KeyNotFoundORKE keyId
+  maybeKey <- getKeyById keyId
 
-  userOwnsKey <- doesUserOwnKeyQuery userId keyId
-  unless userOwnsKey $ throwing_ _UnauthorisedKeyAccessORKE
+  key <- maybe (throwing _KeyNotFoundORKE keyId) pure maybeKey
+  _ <- userOrganisationAuthorisationQuery authUser $ orgPrimaryKeyToGS1CompanyPrefix $ key_org key
```
Comment (sajidanower23) [inline projects/or_scs/src/Mirza/OrgRegistry/Handlers/Keys.hs]: Is this being done just to check that an error is not being thrown?

File: projects/mirza-common-haskell/src/Mirza/Common/GS1BeamOrphans.hs (hunk 1 above)
Comment (a-stacey) [reply projects/mirza-common-haskell/src/Mirza/Common/GS1BeamOrphans.hs]: I wanted to make it clear that this was a DB type specifically.

File: projects/or_scs/src/Mirza/OrgRegistry/Handlers/Keys.hs (hunk 2 above)
Comment (a-stacey) [reply projects/or_scs/src/Mirza/OrgRegistry/Handlers/Keys.hs]: Yeah basically. Specifically the intention of that function is to throw when any of the constraints are violated.
